File: Try_GameEngine5/src/com/example/try_gameengine/scene/MyRect.java
package com.example.try_gameengine.scene;

import com.example.try_gameengine.scene.EasyScene.Type;

import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.RectF;

public class MyRect {
	float x, y, w, h,angle;
	
	Type type;

	public MyRect(float x, float y, float w, float h,Type type) {
		this.x = x;
		this.y = y;
		this.w = w;
		this.h = h;
		
		this.type=type;
	}

	public void setX(float x) {
		this.x = x;
	}

	public void setY(float y) {
		this.y = y;
	}
	
	public float getX()
	{
		return this.x;
	}
	
	public float getY()
	{
		return this.y;
	}

	public void setAngle(float angle) {
		this.angle = angle;
	}
	
	public float getWidth()
	{
		return this.w;
	}
	
	public float getHeight()
	{
		return this.h;
	}
	
	public Type getType()
	{
		return this.type;
	}
	
	public void draw(Canvas canvas, Paint paint) {
		canvas.save();
		canvas.rotate(angle, x+w/2, y+h/2);
		canvas.drawRect(new RectF(x , y  , x + w, y + h), paint);
		canvas.restore();
	}
}

File: Try_GameEngine5/src/com/example/try_gameengine/scene/MyRect.kt
package com.example.try_gameengine.scene

import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.RectF

class MyRect(x: Float, y: Float, w: Float, h: Float, type: EasyScene.Type?) {
    var x: Float
    var y: Float
    var width: Float
    var height: Float
    @JvmField
    var angle: Float = 0f

    var type: EasyScene.Type?

    init {
        this.x = x
        this.y = y
        this.width = w
        this.height = h

        this.type = type
    }

    fun setAngle(angle: Float) {
        this.angle = angle
    }

    fun draw(canvas: Canvas, paint: Paint) {
        canvas.save()
        canvas.rotate(angle, x + this.width / 2, y + this.height / 2)
        canvas.drawRect(RectF(x, y, x + this.width, y + this.height), paint)
        canvas.restore()
    }
}
